initial-promises/node-server/failsafe.cf

```diff
@@ -104,53 +100,60 @@ bundle common g
+      "rudder_tools_files_suse"    slist => { "openvpn-2.2.1-1.x86_64.rpm",
+                                              "openvpn-2.2.1-1.i686.rpm",
+                                              "zypper-repo.tpl",
+                                              "check-zypper-version",
+                                              "checkzmd.pl",
                                             };
     any::
-      "rudder_tools_files"        slist => { @{rudder_tools_files_windows},
-                                             @{rudder_tools_files_android},
-                                             @{rudder_tools_files_unix},
-                                             @{rudder_tools_files_suse},
-                                             "checkroute.pl",
-                                           }, policy => "ifdefined";
+      "rudder_tools_files"         slist => { @{rudder_tools_files_windows},
+                                              @{rudder_tools_files_android},
+                                              @{rudder_tools_files_unix},
+                                              @{rudder_tools_files_suse},
+                                              "checkroute.pl",
+                                            }, policy => "ifdefined";
 
       "uuid"                     string => readfile("${g.uuid_file}", 60);
-      "excludedreps"              slist => { "\.X11", ".*kde.*", "\.svn", "perl" };
+      "excludedreps"             slist  => { "\.X11", ".*kde.*", "\.svn", "perl" };
       "ncf_local_excludedreps"    slist => { "50_techniques", @{g.excludedreps}};
       "rudder_tools_origin"      string => "/var/rudder/tools";
       "rudder_ncf_origin_common" string => "/usr/share/ncf/tree";
       "rudder_ncf_origin_local"  string => "/var/rudder/configuration-repository/ncf";
 
+      "rudder_ncf_hash_file"          string => "ncf_hash_file";
+      "rudder_ncf_origin_common_hash" string => "${rudder_ncf_origin_common}/${rudder_ncf_hash_file}";
+      "rudder_ncf_origin_local_hash"  string => "${rudder_ncf_origin_local}/${rudder_ncf_hash_file}";
+
       "rudder_tools_updated_origin" string => "${rudder_tools_origin}/rudder_tools_updated";
       "rudder_tools_updated"        string => "${rudder_tools}/rudder_tools_updated";
 
+
+      "rudder_tools_updated_filename" string => "rudder_tools_updated";
+      "rudder_tools_updated_origin" string => "${rudder_tools_origin}/${rudder_tools_updated_filename}";
+      "rudder_tools_updated"        string => "${rudder_tools}/${rudder_tools_updated_filename}";
+
+      # The time at which the execution started
+    windows.!cygwin::
+      "execRun"                  string => execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"'{0:yyyy-MM-dd HH:mm:sszzz}' -f (get-date)\"", "noshell");    android::
+      "execRun"                  string => execresult("/system/xbin/date \"+%Y-%m-%d %T+02:00\"", "noshell");
```

Comment:
Something has gone wrong here: there should be a line with "android::" just above this one.